janis_core/tool/test_suite_runner.py

```diff
@@ -0,0 +1,217 @@
+        """
+        # Note, we do not want janis_core to depend on janis_assistant except for when we run these tests
+        # So, we only import this package here in this function
+        # Make sure the correct version of janis_assistant is installed first
+
+        test_helpers.verify_janis_assistant_installed()
+
+        from janis_assistant.main import run_with_outputs
+        from janis_assistant.management.configuration import JanisConfiguration
+
+        output_dir = os.path.join(os.getcwd(), "tests_output", self.tool.id())
+        config = JanisConfiguration(engine=engine)
+        output = run_with_outputs(
+            tool=self.tool, inputs=input, output_dir=output_dir, config=config
+        )
+
+        return output
+
+    def run_one_test_case(self, t: TTestCase, engine):
+        """
+        Run one test case and assert multiple expected output
+        """
+        output = self.run(input=t.input, engine=engine)
+
+        failed = set()
+        succeeded = set()
+
+        for test_logic in t.output:
+            workflow_output = output[test_logic.tag]
+            actual_output = self.get_value_to_compare(test_logic, workflow_output)
+            expected_value = self.get_expected_value(test_logic)
+            test_result = test_logic.operator(actual_output, expected_value)
+
+            if test_result is False:
+                failed.add(
+                    f"{str(test_logic)} {type(expected_value)}"
+                    f" | actual output: {actual_output} {type(actual_output)}"
+                )
+            else:
+                succeeded.add(str(test_logic))
+
+        return failed, succeeded
+
+    def get_expected_value(self, test_logic: TTestExpectedOutput):
+        """
+        Get the expected value of a given expected output.
+        It could be specified in a variable or in a file.
+        """
+        if test_logic.expected_value is not None:
+            return test_logic.expected_value
+        elif test_logic.expected_file is not None:
+            with open(test_logic.expected_file) as expected_file:
+                return expected_file.read()
+        else:
+            raise Exception(
+                "one of `TTestExpectedOutput.expected_value` or `TTestExpectedOutput.expected_file` must not be empty"
+            )
+
+    def get_value_to_compare(
+        self, test_logic: TTestExpectedOutput, output_value: Any
+    ) -> Any:
+        """
+        convert workflow output value to the format we want to compare
+        """
+        output_tag = test_logic.tag
+        output_type = self.tool.outputs_map().get(output_tag).outtype
+        output_value = self._preprocess(
+            test_logic=test_logic,
+            output_value=output_value,
+            output_type=output_type,
+        )
+
+        return self._transform_value(
+            test_logic=test_logic,
+            output_value=output_value,
+            output_type=output_type,
+        )
```

Comment:
We will choose to call `self._transform_value` or the callable passed to this object